doc/mgmt/gnmi/save_on_set.md

```diff
@@ -0,0 +1,220 @@
+
+Currently there are a number of devices that are ‘out in the wild’ that use the current implementation of the UMF and therefore do not persist configuration across reboots and their behavior cannot be changed without changing the configuration tools that interact with them.
+For this reason and for more versatility the save-on-set behavior should be able to be toggled by a command-line parameter to the telemetry executable.
+
+### Requirements
+
+This feature should be off by default to avoid interfering with legacy switches.
+### Architecture Design
+
+This feature does not change the SONiC Architecture
+
+### High-Level Design
+
+- Is it a built-in SONiC feature or a SONiC Application Extension?
+  - built-in SONiC feature
+- What are the modules and sub-modules that are modified for this design?
+  - gNMI Server
+- What are the repositories that would be changed?
+  - [sonic-gnmi](https://github.com/sonic-net/sonic-gnmi)
+  - [sonic-mgmt-common](https://github.com/sonic-net/sonic-mgmt-common)
+- Module/sub-module interfaces and dependencies.
+  - Adds a flag to the gNMI module.
+- SWSS and Syncd changes in detail
+  - N/A
+- DB and Schema changes (APP_DB, ASIC_DB, COUNTERS_DB, LOGLEVEL_DB, CONFIG_DB, STATE_DB)
+  - Adds an entry to the CONFIG_DB to determine if the feature is enabled.
+- Sequence diagram if required.
+![Flow Char](images/save_on_set.png)
+- Linux dependencies and interface
+  - N/A
+- Warm reboot requirements/dependencies
+  - None
+- Fastboot requirements/dependencies
+  - None
+- Scalability and performance requirements/impact
+  - If the feature is enabled, there is performance hit that scales with the size of ConfigDB.
+- Memory requirements
+  - Negligible, a couple function pointers.
+- Docker dependency
+  - None
+- Build dependency if any
+  - None
+- Management interfaces - SNMP, CLI, RestAPI, etc.,
+  - [Configuration and management](#configuration-and-management)
+- Serviceability and Debug (logging, counters, trace etc) related design
+  - Logging of feature being enabled;
+- Is this change specific to any platform? Are there dependencies for platforms to implement anything to make this feature work? If yes, explain in detail and inform community in advance.
+  - Any Platform
+- SAI API requirements, CLI requirements, ConfigDB requirements. Design is covered in following sections.
+
+#### Required Changes
+
+##### Telemetry Executable
+A new command-line parameter will be added to control the behavior of the save-on-set functionality. By default, i.e. when the option is not specified, the gNMI server will behave as it did before this change - the configuration will not be saved to a file without explicit action from the administrator for example by execution of a command via ssh connection.
+
+The new parameter will be: --with-save-on-set and when present it will configure a function pointer variable gnmi.SaveOnSet to point to a function that actually performs the save operation.
```

Comment:
Are we getting this feature enable/disable from config_db field?

Reply:
Added clarification that the startup script will read from the DB to pass the value along to the command line argument on startup.